### PR TITLE
Support type conversion via type(dtype).

### PR DESCRIPTION
**Pull request by gchanan**

### CODE REVIEW

File: tools/autograd/templates/python_variable_methods.cpp [hunk 1]
```diff
@@ -520,26 +520,31 @@ static PyObject * THPVariable_type(PyObject* self, PyObject* args, PyObject* kwa
 {
   HANDLE_TH_ERRORS
   static PythonArgParser parser({
+    "type(Type dtype, bool non_blocking=False)",
```
Comment (colesbury) [inline tools/autograd/templates/python_variable_methods.cpp]: Can you merge this with the overload below? Having only one (non-deprecated) overload gives better error messages

**Review by colesbury — APPROVED** [review]

**Comment by gchanan** [comment]
looks like the windows test is done, but it's not displaying as done.

**Pull request merged**